garak/translator.py

```diff
@@ -0,0 +1,385 @@
+                translated_lines.append(line)
+                continue
+            
+            sentences = re.split(r'(\. |\?)', line.strip())
+            for sentence in sentences:
+                cleaned_sentence = sentence.strip()
+                if cleaned_sentence.isspace() or len(cleaned_sentence) == 0:
+                    continue
+                if cleaned_sentence in {".", "?", ". "}:
+                    continue
+                if cleaned_sentence.replace(".", "") == "":
+                    continue
+                sentence = sentence.lower()
+                translated_line = self._translate(sentence, source_lang=source_lang, target_lang=target_lang)
+                translated_lines.append(translated_line)
+
+        res = '\n'.join(translated_lines)
+
+        return res
+
+
+class EncodingTranslator(SimpleTranslator):
+
+    def _get_response(self, input_text: str, source_lang: Optional[str] = None, target_lang: Optional[str] = None):
+
+        if not (source_lang and target_lang):
+            return input_text
+
+        self._load_translator()
+        translated_lines = []
+
+        split_text = split_input_text(input_text)
+
+        for line in split_text:
+            mean_word_judge = is_english(line, clean_word_option=True)
+
+            if not mean_word_judge:
+                translated_lines.append(line.strip())
+            else:
+                if "$" == line:
+                    translated_lines.append(line)
+                else:
+                    translated_line = self._translate(line, source_lang=source_lang, target_lang=target_lang)
+                    translated_lines.append(translated_line)
+
+        res = '\n'.join(translated_lines)
+
+        return res
+
+    
+class GoodsideTranslator(SimpleTranslator):
+
+    def _get_response(self, input_text: str, source_lang: Optional[str] = None, target_lang: Optional[str] = None):
+        if not (source_lang and target_lang):
+            return input_text
+
+        self._load_translator()
+        translated_lines = []
+
+        for line in input_text.splitlines():
+
+            if contains_invisible_unicode(line):
+                translated_lines.append(line.strip())
+            else:
+                translated_line = self._translate(line, source_lang=source_lang, target_lang=target_lang)
+                translated_lines.append(translated_line)
+
+        res = '\n'.join(translated_lines)
+
+        return res
+    
+
+class LocalTranslator():
+    """Local translation using Huggingface m2m100 models
+    Reference: 
+      - https://huggingface.co/facebook/m2m100_1.2B 
+      - https://huggingface.co/facebook/m2m100_418M
+      - https://huggingface.co/docs/transformers/model_doc/marian
+    """
+
+    def __init__(self, config_root=_config):
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        lang_specs = getattr(config_root.run, 'lang_spec', "jap")  
```

Comment:
`cuda` capability is not a default expectation for this project, while this does fall back to "cpu" there is likely work needed to enable this object on all supported platforms.

`lang_spec` by default based on this PR should just be `"en"` and should not load a model when no translation will ever occur. The translator services should be agnostic of any actual translation and `source_lang` to `target_lang` should just pass thru when both values are equal returning the original values.

Reply:
>cuda capability is not a default expectation for this project, while this does fall back to "cpu" there is likely work needed to enable this object on all supported platforms.

I try to change the default value "cpu". I think it can work if machine does not have GPU.

>lang_spec by default based on this PR should just be "en" and should not load a model when no translation will ever occur.

OK. I change the default value of `lang_spec`. And if the `lang_spec` is "en", it will pass to load a model

>source_lang to target_lang should just pass thru when both values are equal returning the original values.

OK. I add the check function.